superset/views/utils.py

```diff
@@ -184,3 +185,23 @@ def get_datasource_info(datasource_id, datasource_type, form_data):
             'The datasource associated with this chart no longer exists')
     datasource_id = int(datasource_id)
     return datasource_id, datasource_type
+
+
+def apply_display_max_row_limit(sql_results):
+    """
+    Given a `sql_results` nested structure, applies a limit to the number of rows
+
+    `sql_results` here is the nested structure coming out of sql_lab.get_sql_results, it
+    contains metadata about the query, as well as the data set returned by the query.
+    This method limits the number of rows adds a `displayLimitReached: True` flag to the
+    metadata.
+    """
+    display_limit = app.config.get('DISPLAY_MAX_ROW')
+    if (
+        display_limit and
+        sql_results['status'] == QueryStatus.SUCCESS and
+        display_limit < sql_results['query']['rows']
+    ):
+        sql_results['data'] = sql_results['data'][:display_limit]
+        sql_results['displayLimitReached'] = True
+    return sql_results
```

Comment:
Any reason to return `sql_results` here as you're mutating the input? Also could you add Python typing for the inputs/outputs as this helps with readability. See [here](https://github.com/apache/incubator-superset/blob/master/CONTRIBUTING.md#typing) for more detail.

Reply:
I'm returning it so I can call the function inline. And types added!